src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/JsUtilsBuilder.java

```diff
@@ -404,7 +406,11 @@ protected static JMethod addJavaToAttributesMapMethod(JCodeModel jCodeModel, JDe
         final JVar originalParameter = toReturn.param(JMod.FINAL, jCodeModel.ref(Object.class), "original");
         final JBlock block = toReturn.body();
         final JVar mapToReturn = block.decl(JMod.FINAL, narrowedMap, "toReturn", JExpr._new(hashMapField));
-        block.invoke("toAttributesMap").arg(mapToReturn).arg(originalParameter);
+
+        final JConditional nonNull = block._if(jCodeModel.ref(Objects.class).staticInvoke("nonNull").arg(originalParameter));
+        nonNull._then().invoke("toAttributesMap").arg(mapToReturn).arg(originalParameter);
+
+//        block.invoke("toAttributesMap").arg(mapToReturn).arg(originalParameter);
```

Comment:
@manstis 
Overall fine for me - pls just remove the commented line

Reply:
Hahah.. yes looks like I forgot to tidy up my mess.